packages/reactDom/src/ReactServerDOM.ml

```diff
@@ -520,26 +521,34 @@ let rec to_html ~(fiber : Fiber.t) (element : React.element) : (Html.element * j
         context.push debug_info_index (Model.Debug_info_map debug_info_ref);
         ()); *)
       match component () with
-      | element -> to_html ~fiber element)
+      | element -> to_html ~fiber element ~withBodyHtml)
   | Lower_case_element { key; tag; attributes; children } ->
       if fiber.is_root_html_node && tag = "head" then (
         (* in case of a head element, we hoist it to the top of the document, and avoid rendering it in the current node *)
         let html_attributes = ReactDOM.attributes_to_html attributes in
-        let%lwt html_and_json = children |> Lwt_list.map_p (to_html ~fiber) in
+        let%lwt html_and_json = children |> Lwt_list.map_p (to_html ~fiber ~withBodyHtml) in
         let html, model = List.split html_and_json in
         Fiber.push_hoisted_head ~fiber html_attributes html;
         let json = Model.node ~tag:"head" ~key:None ~props:(Model.props_to_json attributes) model in
         Lwt.return (Html.null, json))
       else if fiber.is_root_html_node && is_a_head_child_tag tag then (
+        let children = ReactDOM.moveDangerouslyInnerHtmlAsChildren attributes children in
         let html_props = ReactDOM.attributes_to_html attributes in
-        let%lwt children, model = elements_to_html ~fiber children in
-        let html = Html.node tag html_props [ children ] in
+        let%lwt children_html, children_model = elements_to_html ~fiber children ~withBodyHtml in
+        (* TODO: Could we improve this to avoid the need of normalizing the children model? *)
+        (* elements_to_html always returns a list, and meta/link cannot have children, 
+           if any tag has dangerouslySetInnerHTML, the returned of the moveDangerouslyInnerHtmlAsChildren will be [ InnerHtml innerHtml ] 
+           which is `List [ `Null ] for the model, so we normalize it all `Null *)
+        let normalized_children_model =
+          match children_model with `List [ `Null ] | `List [] -> `Null | _ -> children_model
+        in
+        let html = Html.node tag html_props [ children_html ] in
```

Comment:
Explaining this:
Without the normalize I was having the following error:
<img width="780" alt="Image" src="https://github.com/user-attachments/assets/79c4bed0-64c0-4d06-8bf7-cc3274b6cc5b" />
Because the model had children for the meta/link as you can see here:

<img width="430" alt="Image" src="https://github.com/user-attachments/assets/ef83c35e-2e0e-4337-b30e-6ff95c760460" />

For the style, the issue was because we had both child and dangerouslySetInnerHTML in the model:
<img width="622" alt="Image" src="https://github.com/user-attachments/assets/32d2f18c-28fe-46b9-8907-16bd91e54efb" />

Reply:
It was making the demo fail when there was no body html.

Reply:
Just documenting it here.
That's how arch/server produces a simple link tag:
```
["$","link",null,{"rel":"stylesheet","href":"/output.css"},"$1",[["App","/Users/pedrolisboa/work/server-reason-react/arch/server/render-rsc-to-stream.js",58,6]],1]
```
And that is how server-reason-react does
```
["$","link",null,{"children":[],"rel":"stylesheet","href":"/output.css"},null,[],{}]
```

It looks like we shouldn't have any 'children' value (null or []) for elements that don't have it.